src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        storageDao.add(new User("popova", "justpopova", 26));
+        storageDao.add(new User("sterniuk", "1123313", 30));
+        storageDao.add(new User("pony", "dfyrg6", 17));
+    }
+
+    @Test
+    public void userNull_isNotValid() {
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    public void userSameLogin_isNotValid() {
+        User newUser = new User("popova", "justpopova", 26);
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+
+    @Test
+    public void userLogin_isValid() {
```

Comment:
```suggestion
    public void validLogin_isValid() {
```

Reply:
if everything is okay you need to check if actual user is not null, has id, is the same as the one you passed, and is present in database

Reply:
i don't understand.... do i need just change the name of method or i should change also the logic inside?